app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -211,6 +250,158 @@ def get_locations
           location.match(VALID_LOCATION).to_s
         end
       end
+
+      def parse_stack(deployment)
+        name = deployment.fetch_path('name')
+        uid = resource_uid(@subscription_id,
+                           deployment.fetch_path('resourceGroup'),
+                           TYPE_DEPLOYMENT,
+                           name)
+        child_stacks, resources = find_stack_resources(deployment)
+        new_result = {
+          :type        => ManageIQ::Providers::Azure::CloudManager::OrchestrationStack.name,
+          :ems_ref     => deployment.fetch_path('id'),
+          :name        => name,
+          :description => name,
+          :status      => deployment.fetch_path('properties', 'provisioningState'),
+          :children    => child_stacks,
+          :resources   => resources,
+          :outputs     => find_stack_outputs(deployment),
+          :parameters  => find_stack_parameters(deployment),
+
+          :orchestration_template => find_stack_template(deployment)
+        }
+        return uid, new_result
+      end
+
+      def find_stack_template(deployment)
+        uri = deployment.fetch_path('properties', 'templateLink', 'uri')
+        return unless uri
+
+        content = download_template(uri)
+        return unless content
+
+        get_stack_template(deployment, content)
+        @data_index.fetch_path(:orchestration_templates, deployment.fetch_path('id'))
+      end
+
+      def download_template(uri)
+        require 'open-uri'
+        open(uri) { |f| f.read }
+      rescue => e
+        _log.error("Failed to download Azure template #{uri}. Reason: #{e.inspect}")
+        nil
+      end
+
+      def find_stack_parameters(deployment)
+        raw_parameters = deployment.fetch_path('properties', 'parameters')
+        return [] if raw_parameters.blank?
+
+        stack_id = deployment.fetch_path('id')
+        get_stack_parameters(stack_id, raw_parameters)
+        raw_parameters.collect do |param_key, _val|
+          @data_index.fetch_path(:orchestration_stack_parameters, resource_uid(stack_id, param_key))
+        end
+      end
+
+      def find_stack_outputs(deployment)
+        raw_outputs = deployment.fetch_path('properties', 'outputs')
+        return [] if raw_outputs.blank?
+
+        stack_id = deployment.fetch_path('id')
+        get_stack_outputs(stack_id, raw_outputs)
+        raw_outputs.collect do |output_key, _val|
+          @data_index.fetch_path(:orchestration_stack_outputs, resource_uid(stack_id, output_key))
+        end
```

Comment:
`get_stack_outputs` actually returns the `raw_outputs`, we need to fetch the parsed output object.